### PR TITLE
improve testing

### PR DESCRIPTION
**Pull request by fscherf**
This PR adds some new tests, some new testing helper and documentation.

### CODE REVIEW

File: doc/content/end-user-documentation/testing.rst [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+`lona_project_context <#lona-project-context-example>`_.
+
+`lona_app_context <#lona-app-context-example>`_ spins up a test server using an
+API like the
+`Lona script interface </end-user-documentation/getting-started.html#running-lona-from-a-script>`_,
+`lona_project_context <#lona-project-context-example>`_ takes the project root
+and settings of a
+`Lona project </end-user-documentation/getting-started.html#starting-a-lona-project-from-the-project-template>`_
+to start the test server. Lona test servers run on an unprivileged random port.
+
```
Comment (SmithChart) [inline doc/content/end-user-documentation/testing.rst:18]: In the following examples you use ``playwright`` as browser-interface. But you never introduce it here.  Maybe add it to the list of tools like ``pytest`` and ``pytest-aiohttp``.
Maybe add some words about ``playwright`` vs other browser-testing frameworks here?

Comment (fscherf) [reply doc/content/end-user-documentation/testing.rst:18]: Good point! I added a section about browser testing and a synchronous code example for frameworks like selenium

File: doc/content/end-user-documentation/testing.rst [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+The Lona testing infrastructure is meant to be used together with a browser
+testing library like `playwright <https://playwright.dev/python/docs/intro>`_
+or `selenium <https://selenium-python.readthedocs.io/>`_.  The test suite for
```
Comment (SmithChart) [inline doc/content/end-user-documentation/testing.rst:21]: Nitpicking: Single double space here :-)

File: doc/content/end-user-documentation/testing.rst (hunk 1 above)
Comment (SmithChart) [reply doc/content/end-user-documentation/testing.rst:18]: Thanks! Looks good.

File: doc/content/end-user-documentation/testing.rst (hunk 2 above)
Comment (fscherf) [reply doc/content/end-user-documentation/testing.rst]: Fixed! Thanks!

File: .coveragerc [hunk 3]
```diff
@@ -1,3 +1,10 @@
...
-source = lona
\ No newline at end of file
+source = lona
+
+omit =
+    lona/__main__.py
+    lona/default_routes.py
+    lona/logging.py
+    lona/command_line/*
+    lona/shell/*
```
Comment (laundmo) [inline .coveragerc:10]: im not quite sure why these would absolutely need to be omitted from coverage. testing CLI-args and shel lcommands seems like something that we would want long-term.

Comment (fscherf) [reply .coveragerc]: Hm, you are right. I removed everything but ``lona/default_routes.py`` and ``lona/__main__.py``

**Review by laundmo — APPROVED** [review]
LGTM

**Comment by fscherf** [comment]
@SmithChart, @laundmo: Thanks for reviewing :)

**Pull request merged**